pkg/fanal/artifact/repo/git_test.go

```diff
@@ -199,6 +212,50 @@ func TestArtifact_Inspect(t *testing.T) {
 				SchemaVersion: types.BlobJSONSchemaVersion,
 			},
 		},
+		{
+			name:   "remote repo with userinfo in URL",
+			rawurl: remoteRepoWithCreds,
+			want: artifact.Reference{
+				Name: remoteRepoWithCreds,
+				Type: types.TypeRepository,
+				ID:   "sha256:dc7c6039424c9fce969d3c2972d261af442a33f13e7494464386dbe280612d4c",
+				BlobIDs: []string{
+					"sha256:dc7c6039424c9fce969d3c2972d261af442a33f13e7494464386dbe280612d4c",
+				},
+				RepoMetadata: artifact.RepoMetadata{
+					RepoURL:   ts.URL + "/test-repo.git",
+					Branch:    "main",
+					Tags:      []string{"v0.0.1"},
+					Commit:    "8a19b492a589955c3e70c6ad8efd1e4ec6ae0d35",
+					CommitMsg: "Update README.md",
+					Author:    "Teppei Fukuda <knqyf263@gmail.com>",
+					Committer: "GitHub <noreply@github.com>",
+				},
+			},
```

Comment:
Most fields are already tested in other test cases, and comparing them here again is just redundant. What if adding `TestArtifact_InspectWithUserInfo` to compare only the repo URL?

Reply:
something like this?
https://github.com/aquasecurity/trivy/pull/9391/commits/dac0d88292db6e32c015be2b0ec7bbdb65b0d0bf